lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -306,6 +311,20 @@ class FastAPIHistoryContents:
         "/api/histories/{history_id}/contents",
         name="history_contents",
         summary="Returns the contents of the given history.",
+        responses={
+            200: {
+                "model": List[AnyHistoryContentItem],
+                "description": (
+                    "When using `v=dev&view=count` the `total_matches` of the search is returned in the headers"
+                ),
+                "headers": {
+                    "total_matches": {
+                        "description": "The number of items that match the search query",
+                        "schema": {"type": "integer"},
+                    },
+                },
+            },
+        },
```

Comment:
Can you use a custom media type instead of `v=dev&view=count` ? We have to stop propagating these legacy behaviors.

Reply:
Sure I can try!
Is setting the request content type to something like `header["content-type"] = "application/json-count"` and checking the header in the endpoint what you had in mind? Or I'm misunderstanding?

To be honest, it feels a bit weird to me messing with the content type, should we just add a different optional header or just add a new optional parameter to the endpoint?

By the way, what about the naming convention for headers? should we rename `total_matches` to `total-matches` or doesn't matter?